piksi_multi_cpp/src/receiver/receiver_factory.cc

```diff
@@ -88,6 +90,23 @@ ReceiverFactory::ReceiverType ReceiverFactory::inferType(
     const Device::Ptr& dev) {
   if (!dev.get()) return ReceiverType::kUnknown;
 
+  // Create settings object.
+  SettingsIo settings_io(ros::NodeHandle(), dev);
```

Comment:
Thinking about this again, I came to the conclusion that there should be a separate simple factory method to create the settingsio. Why? If we have a factorymethod that creates the receivers, whoever uses that methos should not need to know what type of receiver is returned getting. Otherwise you have to do pointer casting from base to subclass (a code smell-it shows that some abstraction is not consistent)